src/newtondynamics_physics/SysNewton.cpp

```diff
@@ -0,0 +1,428 @@
+{
+    SPDLOG_LOGGER_INFO(rScene.get_application().get_logger(),
+                      "Initing Sysnewton");    rScene.debug_update_add(rScene.get_update_order(), "physics", "wire", "",
+                            &SysNewton::update_world);    //NewtonWorldSetUserData(m_nwtWorld, this);
+
+    // Connect signal handlers to destruct Newton objects when components are
+    // deleted.
+
+    rScene.get_registry().on_destroy<ACompNwtBody>()
+                    .connect<&SysNewton::on_body_destruct>();
+
+    rScene.get_registry().on_destroy<ACompNwtCollider>()
+                    .connect<&SysNewton::on_shape_destruct>();
+
+    rScene.get_registry().on_destroy<ACompNwtWorld>()
+                    .connect<&SysNewton::on_world_destruct>();
+}
+
+void SysNewton::update_world(ActiveScene& rScene)
+{
+    ActiveReg_t &rReg = rScene.get_registry();
+
+    // Try getting the newton world from the scene root.
+    // It is very unlikely that multiple physics worlds are ever needed
+    auto *worldComp = rReg.try_get<ACompNwtWorld>(rScene.hier_get_root());
+
+    if (worldComp == nullptr)
+    {
+        return; // No physics world component
+    }
+
+    if (worldComp->m_nwtWorld == nullptr)
+    {
+        // Create world if not yet initialized
+        worldComp->m_nwtWorld = NewtonCreate();
+        NewtonWorldSetUserData(worldComp->m_nwtWorld, &rScene);
+    }
+
+    NewtonWorld const* nwtWorld = worldComp->m_nwtWorld;
+
+    auto viewBody = rScene.get_registry().view<ACompRigidBody>();
+
+    // Iterate all rigid bodies in scene
+    for (ActiveEnt ent : viewBody)
+    {
+        auto &entBody = viewBody.get<ACompRigidBody>(ent);
+
+        // temporary: delete Newton Body if something is dirty
+        if (entBody.m_colliderDirty)
+        {
+            rReg.remove_if_exists<ACompNwtBody>(ent);
+            entBody.m_colliderDirty = false;
+        }
+
+        // Initialize body if not done so yet;
+        if (! rReg.any_of<ACompNwtBody>(ent))
+        {
+            create_body(rScene, ent, nwtWorld);
+        }
+
+        // Recompute inertia and center of mass if rigidbody has been modified
+        if (entBody.m_inertiaDirty)
+        {
+            compute_rigidbody_inertia(rScene, ent);
+            SPDLOG_LOGGER_TRACE(m_scene.get_application().get_logger(), "Updating RB : new CoM Z = {}",
+                              entBody.m_centerOfMassOffset.z());
+        }
+
+    }
+
+    // Update the world
+    NewtonUpdate(nwtWorld, rScene.get_time_delta_fixed());
+
+    auto viewNwtBody = rScene.get_registry().view<ACompNwtBody>();
+    auto viewTf = rScene.get_registry().view<ACompTransform>();
+
+    // Apply transform changes after the Newton world(s) updates
+    for (ActiveEnt ent : viewBody)
+    {
+        auto &entNwtBody      = viewNwtBody.get<ACompNwtBody>(ent);
+        auto &entTransform    = viewTf.get<ACompTransform>(ent);
+
+        float mass, x, y, z;
+        NewtonBodyGetMass(entNwtBody.body(), &mass, &x, &y, &z);
```

Comment:
I have a vague feeling that this should be a different "system", as if there *ARE* more than one physics world, we would want to apply the transformations after all of the world have updated.

Reply:
True. Updating the physics world itself also doesn't modify ACompTransform, so separating into its own system is a good idea.